crates/ruff_linter/resources/test/fixtures/ruff/for_variable_used_after_block.py

```diff
@@ -0,0 +1,81 @@
+for global_var in []:
+    _ = global_var
+    pass
+
+# ❌
+_ = global_var
+
+def foo():
+    # For control var used outside block
+    for event in []:
+        _ = event
+        pass
+
+    # ❌
+    _ = event
+
+    for x in []:
+        _ = x
+        pass
+    
+    # Using the same control variable name in a different loop is ok
+    for x in []:
+        _ = x
+        pass
+
+    for y in []:
+        # ❌ x is used outside of the loop it was defined in (meant to use y)
+        if x == 5:
+            pass
+
+    # Assign a variable before the loop
+    room_id = 3
+    _ = room_id
+    # Use the same variable name in a loop
+    for room_id in []:
+        _ = room_id
+        pass
+
+    # ❌ After the loop is not ok because the value is probably not what you expect
+    _ = room_id
+
+    # ❌ Augmented assignment is not allowed because the value is probably not what you expect
+    room_id += 1
+
+    # Assigning again after the loop is ok
+    room_id = 5
+    room_id += 1
+    _ = room_id
+
```

Comment:
#### Delete statements

> We may explicitly want to allow usages in delete statements ([example](https://github.com/pandas-dev/pandas/blob/0cdc6a48302ba1592b8825868de403ff9b0ea2a5/pandas/__init__.py#L19))

Seems harmless to allow :+1:

Reply:
Usages in `del` statements are just as problematic as other usages, since they also result in `NameError`s if the variable is undefined:

```pycon
>>> iterable = []
>>> for item in iterable:
...     pass
... 
>>> del item
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'item' is not defined. Did you mean: 'iter'?
```